google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/ValidateTextAd.java

```diff
@@ -0,0 +1,178 @@
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new ValidateTextAd().runExample(googleAdsClient, params.customerId, params.adGroupId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupId the ad group ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adGroupId) {
+    String adGroupResourceName = ResourceNames.adGroup(customerId, adGroupId);
+
+    // Creates the expanded text ad info.
+    ExpandedTextAdInfo expandedTextAdInfo =
+        ExpandedTextAdInfo.newBuilder()
+            .setDescription(StringValue.of("Luxury Cruise to Mars"))
+            .setHeadlinePart1(StringValue.of("Visit the Red Planet in style"))
+            // Adds a headline that will trigger a policy violation to demonstrate error handling.
+            .setHeadlinePart2(StringValue.of("Low-gravity fun for everyone!!"))
+            .build();
+
+    // Wraps the info in an Ad object.
+    Ad ad =
+        Ad.newBuilder()
+            .setExpandedTextAd(expandedTextAdInfo)
+            .addFinalUrls(StringValue.of("http://www.example.com"))
+            .build();
+
+    // Builds the final ad group ad representation.
+    AdGroupAd adGroupAd =
+        AdGroupAd.newBuilder()
+            .setAdGroup(StringValue.of(adGroupResourceName))
+            .setStatus(AdGroupAdStatus.PAUSED)
+            .setAd(ad)
+            .build();
+
+    AdGroupAdOperation operation = AdGroupAdOperation.newBuilder().setCreate(adGroupAd).build();
+
+    try (AdGroupAdServiceClient adGroupAdServiceClient =
+        googleAdsClient.getLatestVersion().createAdGroupAdServiceClient()) {
+      // Issues the mutate request setting validateOnly=true.
+      MutateAdGroupAdsResponse response =
+          adGroupAdServiceClient.mutateAdGroupAds(
+              Long.toString(customerId), ImmutableList.of(operation), false, true);
+
+      // Since validation is ON, result will be null.
```

Comment:
Is this code effectively unreachable? If so should we not flag this as an error, e.g. "Expanded text ad passed validation, this is unexpected."

Reply:
This code is unreachable in the current state. However, if you were to remove the policy error in the headline, the sample will print "Expanded text ad validated successfully." which I think is the appropriate behavior.